app/src/main/java/com/example/myapplication/FirstFragment.java

```diff
@@ -115,16 +124,33 @@ public void afterTextChanged(Editable s) {
                 if (input != null) {
                     String matchNumber = String.valueOf(input);
                     UserModel.getMatchData().setMatchNumber(matchNumber);
+                    try {
+                        binding.input.setText(MainActivity.teams.get(Integer.parseInt(matchNumber) - 1));
+                    } catch (IndexOutOfBoundsException e){
+                        if (matchNumber.equals("0")) {
```

Comment:
Can you check for this condition ahead of time in order to avoid the exception? Exceptions should generally be avoided when possible, both for performance and cleanliness.